### PR TITLE
Adding a module to write metrics to InfluxDb

### PR DESCRIPTION
**Pull request by leelasharma**
Adding metrics-influxdb module to enable writing application metrics to [influxdb] (https://influxdb.com/docs/v0.9/introduction/overview.html). 
The latest v0.9 influxdb release was announced recently and it will be great to have metrics report to influxdb as well, similar to graphite.

### CODE REVIEW

**Comment by cemo** [comment]
@ryantenney I know you are reluctant for new modules but InfluxDb usage is greatly increased. Adding support for InfluxDB would be nice. 

:+1:

**Comment by ryantenney** [comment]
Given the newfound focus in v4 on tagged metrics, I will be accepting new reporters for backends which support tags.

**Comment by leelasharma** [comment]
@ryantenney That's great to know. This metrics-influxdb module also supports tags.

**Comment by ryantenney** [comment]
@leelasharma merged, thanks!

**Comment by leelasharma** [comment]
@ryantenney @cemo  Actually no real reason, thanks for pointing out. I can submit a pull request to use the same jackson libraries as the project.

File: metrics-influxdb/src/main/java/com/codehale/metrics/influxdb/InfluxDbHttpSender.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+     * @param hostname      the influxDb hostname
+     * @param port          the influxDb http port
+     * @param database      the influxDb database to write to
+     * @param authString    the authorization string to be used to connect to InfluxDb, of format username:password
+     * @param timePrecision the time precision of the metrics
+     * @throws Exception exception while creating the influxDb sender(MalformedURLException)
+     */
+    public InfluxDbHttpSender(final String hostname, final int port, final String database, final String authString,
+                              final TimeUnit timePrecision) throws Exception {
+        this.url = new URL("http", hostname, port, "/write");
```
Comment (mwuertinger) [inline metrics-influxdb/src/main/java/com/codehale/metrics/influxdb/InfluxDbHttpSender.java]: We should either add a flag to support HTTPS or we should offer a constructor allowing to pass in the full URL.

**Comment by ryantenney** [comment]
Agreed. I am planning on rewriting this to use the Apache http client. I'll make certain it supports https.

**Comment by mnuessler** [comment]
In general HTTPS support makes perfect sense, but InfluxDB 0.9 does currently not support HTTPS. There is an [open ticket](https://github.com/influxdb/influxdb/issues/2115) and it looks like it will be included in the next minor release (0.9.2).

For the rewrite it could make sense to have a look at the official [InfluxDB Java client|https://github.com/influxdb/influxdb-java] too. Seems that they are close to releasing a new version that supports the new InfluxDB 0.9 API.

**Comment by jhorwit2** [comment]
I'd suggest also looking at line protocol versus json as it's the preferred method and not deprecated like JSON is. With line protocol there are some performance notes to remember, like sorting tags & escaping certain chars.

**Comment by mnp** [comment]
So what's the status on this feature? Is anyone working on it?

**Comment by arteam** [comment]
https://github.com/davidB/metrics-influxdb and https://github.com/iZettle/dropwizard-metrics-influxdb are the most up to date InfluxDB reporters for Metrics 3.2. Unfortunately, for the 4.0 version there is no active maintained reporter under the `io.dropwizard.metrics` package.

**Comment by mnp** [comment]
Okay thanks for the update!

**Pull request merged**